src/java.base/share/classes/sun/security/provider/certpath/RevocationChecker.java

```diff
@@ -83,23 +83,55 @@
         String ocspSubject;
         String ocspIssuer;
         String ocspSerial;
+        boolean ocspNonce;
     }
+    private RevocationProperties rp;
+    private static final int DEFAULT_NONCE_BYTES = 16;
 
     RevocationChecker() {
+        rp = getRevocationProperties();
         legacy = false;
+        setDefaultNonce();
     }
 
     RevocationChecker(TrustAnchor anchor, ValidatorParams params)
         throws CertPathValidatorException
     {
+        rp = getRevocationProperties();
         legacy = true;
+        setDefaultNonce();
         init(anchor, params);
     }
 
+    private void setDefaultNonce() {
+        byte[] nonce = null;
+
+        // Set the nonce by default in OCSP request extension when the sytem property
+        // jdk.security.certpath.ocspNonce=true.
+        if (rp.ocspNonce) {
+            try {
+                List<Extension> tmpExtensions = Collections.<Extension>emptyList();
+                Extension nonceExt = new OCSPNonceExtension(DEFAULT_NONCE_BYTES);
+
+                tmpExtensions = new ArrayList<Extension>();
+                tmpExtensions.add(nonceExt);
+                setOcspExtensions(tmpExtensions);
```

Comment:
It seems like you could collapse 113 - 118 into something like:
setOcspExtensions(List.of(new OCSPNonceExtension(DEFAULT_NONCE_BYTES)));

At the very least, it looks like you could do away with 113, since you immediately change the value of the tmpExtensions reference on 116.

Reply:
collapsing done.